### PR TITLE
hw jdbc

### PR DESCRIPTION
**Pull request by Taniann**

### CODE REVIEW

File: src/main/java/ua/mateacademy/tania/lesson17/jdbc/Office.java [hunk 1]
```diff
@@ -85,4 +86,20 @@ public String toString() {
...
+        return Objects.equals(id, office.id) &&
+                Objects.equals(city, office.city) &&
+                Objects.equals(region, office.region);
+    }
+
+    @Override
+    public int hashCode() {
+
+        return Objects.hash(id, city, region);
+    }
```
Comment (MateAcademy) [inline src/main/java/ua/mateacademy/tania/lesson17/jdbc/Office.java:104]: You can not do these methods, they are not needed here

File: src/main/java/ua/mateacademy/tania/lesson17/jdbc/OfficeDaoImpl.java [hunk 2]
```diff
@@ -45,6 +42,70 @@
...
+                "INSERT INTO offices (office, city, region, mgr, target, sales ) VALUES (?,?,?,?,?,?) ");
+
+        preparedStatement.setBigDecimal(1, new BigDecimal(office.getId()));
+        preparedStatement.setString(2, office.getCity());
+        preparedStatement.setString(3, office.getRegion());
+        preparedStatement.setBigDecimal(4, new BigDecimal(office.getMgr()));
+        preparedStatement.setBigDecimal(5, office.getTarget());
+        preparedStatement.setBigDecimal(6, office.getSale());
+
+        Boolean isRowInserted = preparedStatement.executeUpdate() > 0;
```
Comment (MateAcademy) [inline src/main/java/ua/mateacademy/tania/lesson17/jdbc/OfficeDaoImpl.java]: Why so confusing you write, it is difficult to understand this line.

**Review by MateAcademy — APPROVED** [review]
Very good homework

File: src/main/java/ua/mateacademy/tania/lesson17/jdbc/OfficeDaoImpl.java [hunk 3]
```diff
@@ -45,6 +42,70 @@
...
+        PreparedStatement preparedStatement = connection.prepareStatement("SELECT* FROM offices WHERE office=?");
+        preparedStatement.setBigDecimal(1, new BigDecimal(id));
+        ResultSet rs = preparedStatement.executeQuery();
+
+        Office office = null;
+        if (rs.next()) {
+            office = extraxtFromResultSet(rs);
+        }
+
+        rs.close();
```
Comment (SerhiiPasko) [inline src/main/java/ua/mateacademy/tania/lesson17/jdbc/OfficeDaoImpl.java]: Close it in finally block. If you have some exception this is don't work. In other also

**Review by SerhiiPasko — CHANGES_REQUESTED** [review]
Read comment

File: src/main/java/ua/mateacademy/tania/lesson17/jdbc/Office.java (hunk 1 above)
Comment (Taniann) [reply src/main/java/ua/mateacademy/tania/lesson17/jdbc/Office.java:104]: > You can not do these methods, they are not needed here

I need it for testing

**Review by SerhiiPasko — APPROVED** [review]
Good

**Pull request merged**